### PR TITLE
Redux Toolkit

### PR DESCRIPTION
**Pull request by lusbenjamin**
### Motivation
We are about to convert all of the remaining application state into Redux (I hope). Adding [Redux Toolkit](https://redux-toolkit.js.org/) to our stack before we do this is probably a very good idea in terms of reducing potential mistakes and using best practices from the start.

This PR:
- adds the `redux-toolkit` package dependency
- sets up `configureStore` in the display processes
- converts `topNav` into a slice to demonstrate `createSlice`
- refactors the Explore page state into a proper slice
  - bugfix: `activeEvents` should be an Array
  - upgrade: better logic for determining when a query has no more results

Highly recommended reading:
- https://redux-toolkit.js.org/tutorials/basic-tutorial
- https://redux-toolkit.js.org/usage/usage-with-typescript

### CODE REVIEW

**Comment by Manuel-777** [comment]
I'm digging this, One of the reasons I made the `dispatchAction` function was this, and wasnt sure if it was correct at all to "interfere" in the default flow of Redux like that :D

**Comment by lusbenjamin** [comment]
As 1 of the 3 major 3rd-party packages (Redux, React-Redux, Redux Toolkit), these conventions are highly likely to futureproof / understood by new contributors familiar w Redux. (Also Redux Toolkit is part of Create React App.)

File: src/window_main/app/ipcListeners.tsx [hunk 1]
```diff
@@ -165,11 +166,7 @@ export default function ipcListeners(dispatcher: unknown): void {
   });
 
   ipc.on("settings_updated", (): void => {
-    dispatchAction(
-      dispatcher,
-      SET_TOP_NAV,
-      pd.settings.last_open_tab || MAIN_HOME
-    );
+    dispatcher(setTopNav(pd.settings.last_open_tab ?? MAIN_HOME));
```
Comment (Manuel-777) [inline src/window_main/app/ipcListeners.tsx]: Maybe I just dont understand the null coalescing properly, but shouldn't `pd.settings.last_open_tab` be a null for it to return `MAIN_HOME` as a default? currently it cant be other than a number or undefined.

Comment (lusbenjamin) [reply src/window_main/app/ipcListeners.tsx]: coalescing handles both `null` and `undefined`: https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#nullish-coalescing

File: src/shared/redux/reducers.tsx [hunk 2]
```diff
@@ -351,18 +351,6 @@ const shareDialog = (
   }
 };
 
-const activeEvents = (
-  state: string[] = defaultState.activeEvents,
-  action: Action
-): string[] => {
-  switch (action.type) {
-    case SET_ACTIVE_EVENTS:
-      return { ...state, ...action.value };
```
Comment (lusbenjamin) [inline src/shared/redux/reducers.tsx]: this was a bug; the intended return value was probably `return [ ...state, ...action.value ];`

Comment (Manuel-777) [reply src/shared/redux/reducers.tsx]: Ohhh, great catch! I think this was the source of the array conversion issues I was having !

**Comment by Manuel-777** [comment]
My attempt at learning this was sucessful! Should we go and convert the whole reducer here or in separate PRs?

**Pull request merged**